src/coreclr/jit/codegencommon.cpp

```diff
@@ -5550,6 +5585,7 @@ void CodeGen::genFnProlog()
         if ((intRegState.rsCalleeRegArgMaskLiveIn | floatRegState.rsCalleeRegArgMaskLiveIn) != RBM_NONE)
         {
             genHomeRegisterParams(initReg, &initRegZeroed);
+            genHomeStackPartOfSplitParameter(initReg, &initRegZeroed);
```

Comment:
```suggestion
            genHomeStackPartOfSplitParameter(initReg, &initRegZeroed);
            genHomeRegisterParams(initReg, &initRegZeroed);
```

Reply:
I think you should actually put this above and outside the if. If you enabled independent promotion of these parameters then it would be possible for the register part to be dead without the stack part being dead.